src/modules/adaptors/Compound/CTokenAdaptor.sol

```diff
@@ -192,16 +239,56 @@ contract CTokenAdaptor is BaseAdaptor {
      * @notice Allows strategists to withdraw assets from Compound.
      * @param market the market to withdraw from.
      * @param amountToWithdraw the amount of `market.underlying()` to withdraw from Compound
+     * NOTE: `redeem()` is used for redeeming a specified amount of cToken, whereas `redeemUnderlying()` is used for obtaining a specified amount of underlying tokens no matter what amount of cTokens required.
+     * NOTE: Purposely allowed withdrawals even while 'IN' market for this strategist function.
      */
     function withdrawFromCompound(CErc20 market, uint256 amountToWithdraw) public {
-        _validateMarketInput(address(market));
-
         uint256 errorCode;
         if (amountToWithdraw == type(uint256).max) errorCode = market.redeem(market.balanceOf(address(this)));
         else errorCode = market.redeemUnderlying(amountToWithdraw);
 
+
         // Check for errors.
         if (errorCode != 0) revert CTokenAdaptor__NonZeroCompoundErrorCode(errorCode);
+
+        uint256 hf = _getHealthFactor(address(this), comptroller);
```

Comment:
This is left here currently as I am troubleshooting tests `testBorrowInSameCollateralMarket` and `testStrategistWithdrawTooLowHF` in `CompoundV2AdditionalTests.t.sol`.

I was comparing the HF calculated here vs the methods I created within the test file itself. It all shows that the HF is calculated the same way. The problem is the `amountToWithdraw` calculated using the helper. That seems to be incorrect but I keep looking at it and can't seem to get atop of what is incorrect.